### ISSUE TITLE
mutable_index pattern

### ISSUE THREAD

**Issue by yegor256** [yegor256/aibolit#147]
Here it is:

```java
class Foo {
  void bar() {
    for (int i = 0; i < 10; i++) {
      i++; // here
    }
  }
}
```

Once we see that the index variable of the `for` is being modified in the body of the loop, it's a pattern. It's a bad practice.

**Comment by 0crat** [comment]
@acheshkov[/z](https://www.0crat.com/u/acheshkov) please, pay attention to this issue

**Comment by acheshkov** [comment]
@0crat in

**Comment by 0crat** [comment]
The job [#147](https://github.com/yegor256/aibolit/issues/147) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/0f4d3996-92ec-442c-8b8b-7c9db86c1e5e); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@0crat wait for #240 merge

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/aibolit/issues/147#issuecomment-621869270))

@paulodamaso This is what I know about this job in CT5UB7XAP, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#147](https://github.com/yegor256/aibolit/issues/147) is [in scope](http://www.zerocracy.com/policy.html#14) for 20days
  * The role is `DEV`
  * The job is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) for [7days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get 50% of their hourly rate for this job
  * The job has an [impediment](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @yegor256[/z](https://www.0crat.com/u/yegor256): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT5UB7XAP?q=%7Bjob%3A%27gh%3Ayegor256%2Faibolit%23147%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by paulodamaso** [comment]
@acheshkov I think that it is still in scope, please remove it (0crat out)

**Comment by acheshkov** [comment]
@0crat out

**Comment by 0crat** [comment]
@acheshkov[/z](https://www.0crat.com/u/acheshkov) please, pay attention to this issue

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/aibolit/issues/147#issuecomment-622250199))

@acheshkov GitHub issue is still open, won't close

**Comment by paulodamaso** [comment]
@acheshkov Please close

**Comment by paulodamaso** [comment]
@acheshkov ping

**Comment by paulodamaso** [comment]
@acheshkov Please take a look at this one

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#283](https://github.com/yegor256/aibolit/issues/283) is still not solved.

**Comment by paulodamaso** [comment]
@acheshkov ping

**Comment by paulodamaso** [comment]
@acheshkov ping

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/aibolit/issues/147#issuecomment-633271357))

@paulodamaso This is what I know about this job in CT5UB7XAP, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#147](https://github.com/yegor256/aibolit/issues/147) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT5UB7XAP?q=%7Bjob%3A%27gh%3Ayegor256%2Faibolit%23147%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by acheshkov** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/aibolit/issues/147#issuecomment-633604605))

@acheshkov Job `gh:yegor256/aibolit#147` is not assigned, can't get performer

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#487](https://github.com/cqfn/aibolit/issues/487) is still not solved.

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#743](https://github.com/cqfn/aibolit/issues/743) is still not solved; solved: [#487](https://github.com/cqfn/aibolit/issues/487).

**Comment by 0pdd** [comment]
@yegor256 all 2 puzzles are solved here: [#487](https://github.com/cqfn/aibolit/issues/487), [#743](https://github.com/cqfn/aibolit/issues/743).
